#	Multiple Linear Regression

Housing price prediction using the Boston Housing Dataset by capturing linear relationships between multiple input features and the target value.

In [10]:
import numpy as np
import pandas as pd

from tensorflow.keras.datasets import boston_housing

from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [2]:
(X_train, y_train), (X_test, y_test) = boston_housing.load_data()

57026/57026 ━━━━━━━━━━━━━━━━━━━━ 0s 1us/step


In [11]:
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

In [12]:
def evaluate_model(name, y_true, y_pred):
    mse = mean_squared_error(y_true, y_pred)
    r2 = r2_score(y_true, y_pred)

    print(f"\n{name}")
    print(f"MSE: {mse:.4f}")
    print(f"R² Score: {r2:.4f}")

In [13]:
def multiple_linear_regression(X_train, X_test, y_train, y_test):
    model = LinearRegression()
    model.fit(X_train, y_train)

    y_pred = model.predict(X_test)
    evaluate_model("Multiple Linear Regression", y_test, y_pred)

In [14]:
multiple_linear_regression(X_train, X_test, y_train, y_test)


Multiple Linear Regression
MSE: 23.1956
R² Score: 0.7214


The model explains about 72.14% of the variation in housing prices (R² = 0.7214) and has a much lower error of 23.20 (MSE), showing a significant improvement in prediction accuracy with multiple features.